src/main/java/com/github/sdonkov/PrefixTree.java

```diff
@@ -75,7 +80,115 @@ public boolean contains(CharSequence word) {
         return false;
     }
 
-    public boolean add(CharSequence word) {
+    @Override
+    public int size() {
+        return countNodes(root);
+    }
+
+    private int countNodes(Node node) {
+        int count = node.isEndOfWord() ? 1 : 0;
+        for (Node child : node.edges) {
+            count += countNodes(child);
+        }
+        return count;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size() == 0;
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        if (!(o instanceof CharSequence)) {
+            return false;
+        }
+        CharSequence word = (CharSequence) o;
+        TraverseResult traverseResult = traverse((T) word);
+        if (traverseResult.index == word.length() && traverseResult.node.isEndOfWord()) {
+            LOGGER.trace("Tree contains this word - {}", word);
+            return true;
+        }
+        LOGGER.trace("Tree doesn't contain this word - {}", word);
+        return false;
+    }
+
+    public Iterator<T> iterator() {
+        return new Iterator<T>() {
+            private final ArrayList<T> words = new ArrayList<>();
+            private int index = 0;
+
+            {
+                collectWords(root, new StringBuilder());
+            }
+
+            private void collectWords(Node node, StringBuilder word) {
+                if (node.isEndOfWord()) {
+                    words.add((T) word.toString());
+                }
+                for (Node edge : node.edges) {
+                    word.append(edge.value);
+                    collectWords(edge, word);
+                    word.deleteCharAt(word.length() - 1);
+                }
+            }
+
+            @Override
+            public boolean hasNext() {
+                return index < words.size();
+            }
+
+            @Override
+            public T next() {
+                if (index >= words.size()) {
+                    throw new NoSuchElementException();
+                }
+                return words.get(index++);
+            }
+
+            @Override
+            public void remove() {
+                if (index <= 0) {
+                    throw new IllegalStateException();
+                }
+                PrefixTree.this.remove(words.remove(--index));
```

Comment:
Here and below, let's drop support for removal for now. It will make things a lot more complicated and is usually not in the use-cases for tries (dictionaries, spell-checking, etc.).